packages/visual-editor/src/components/Dropzone/Dropzone.tsx

```diff
@@ -125,6 +128,22 @@ export function Dropzone({
     node?.data.blockId
   );
 
+  // To avoid a circular dependency, we create the recursive rendering function here and trickle it down
+  const renderDropzone = useCallback(
+    (node: CompositionComponentNode, props?: Record<string, unknown>) => {
```

Comment:
Should explicitly type this to the `RenderDropZone` function instead of relying on duck typing.